app/assets/javascripts/weaknesses.js

```diff
@@ -2,8 +2,16 @@ $(document).on('change', '[data-mark-impact-as]', function () {
   var impact = $(this).data('markImpactAs')
   var markOn = $(this).data('markImpactOn')
 
-  if ($(this).val() === markOn)
-    $('[id$=_impact_' + impact.toLowerCase() + ']').prop('checked', true)
+  if ($(this).val() === markOn) {
+    if (isNaN(impact)) {
```

Comment:
Yo no se como llegás a estas funciones, es muy raro que se use algo así excepto cuando vas a usar la variable en una operación numérica. Podés hacer:
```suggestion
    if (!impact) {
```

Que es casi lo mismo, pero después se hace un .toLowerCase, o sea, ni en pedo eso debería entrar si el valor es NaN (y menos si es !impact), debería ser:

```suggestion
    if (impact) {
```
En todo caso ¿Lo probaste? Quiero videos la próxima.